src/js/components/RangeInput/StyledRangeInput.js

```diff
@@ -1,14 +1,75 @@
+import { rgba } from 'polished';
 import styled, { css } from 'styled-components';
 
 import { focusStyle, normalizeColor, parseMetricToNum } from '../../utils';
 import { defaultProps } from '../../default-props';
 
+// opacity of the bound trumps the track opacity
+const getBoundOpacity = (props, bound) => {
+  return props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    props.theme.rangeInput.track[bound] &&
+    props.theme.rangeInput.track[bound].opacity
+    ? props.theme.rangeInput.track[bound].opacity
+    : 1;
+};
+
+const getBoundColor = (props, bound) => {
+  if (
+    props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    props.theme.rangeInput.track[bound] &&
+    props.theme.rangeInput.track[bound].color
+  ) {
+    return rgba(
+      normalizeColor(props.theme.rangeInput.track[bound].color, props.theme),
+      getBoundOpacity(props, bound),
+    );
+  }
+  // If bound color is undefined pick the default track color with bound opacity
+  return rgba(
+    normalizeColor(props.theme.rangeInput.track.color, props.theme),
+    getBoundOpacity(props, bound),
+  );
+};
+
+const trackColorStyle = props => {
+  // backward compatibility in case no bounds are defined
+  if (
+    props.theme.rangeInput &&
+    props.theme.rangeInput.track &&
+    !props.theme.rangeInput.track.lower &&
+    !props.theme.rangeInput.track.upper
+  ) {
+    const color = rgba(
+      normalizeColor(props.theme.rangeInput.track.color, props.theme),
+      0.2,
+    );
+    if (color === 'rgba(0,0,0,0.2)') return `background: ${color}`;
+  }
+
+  // 'max' defaults to 100 in case not specified
+  const max = props.max || 100;
+  const thumbPosition = `${(props.value / max) * 100}%`;
```

Comment:
I think we also need to handle a `min` value.

Reply:
In what way? Not sure what you mean.

Reply:
If `min={10} max={20} value={15}` then `thumbPosition === '75%'` when it should be `'50%'.

Reply:
Good call! thank you

Reply:
Done.